packages/app/src/entities/currency-rates/context/provider.tsx

```diff
@@ -0,0 +1,30 @@
+'use client'
+
+import React, { useState, useEffect } from 'react'
+import { CurrencyRates } from '../types'
+import { CurrencyRatesContext } from './context'
+import { getCryptoCurrenciesRates } from '../api/currency-rates'
+
+const UPDATE_RATES_INTERVAL_MS = 60_000
+
+export function CurrencyRatesProvider({ children }: { children: JSX.Element }) {
+  const [currencyRates, setCurrencyRates] = useState<CurrencyRates>()
+
+  const getRates = async () => {
+    const apiRates = await getCryptoCurrenciesRates()
+    if (apiRates) setCurrencyRates(apiRates)
+  }
+
+  useEffect(() => {
+    getRates()
+    setInterval(() => {
+      getRates()
+    }, UPDATE_RATES_INTERVAL_MS)
+  }, [])
```

Comment:
We have to clear interval on unmount.

```suggestion
  useEffect(() => {
    getRates()
    const intervalId = setInterval(() => {
      getRates()
    }, UPDATE_RATES_INTERVAL_MS)
    return () => {
      clearInterval(intervalId)
    }
  }, [])
```

Reply:
added